x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/streams/group_stream.ts

```diff
@@ -45,7 +45,49 @@ export class GroupStream extends StreamActiveRecord<Streams.GroupStream.Definiti
 
     this._definition = definition;
 
-    return { cascadingChanges: [], changeStatus: 'upserted' };
+    const missingMembers: string[] = [];
+    for (const member of this._definition.group.members) {
+      if (!desiredState.has(member)) {
+        missingMembers.push(member);
+      }
+    }
+
+    const cascadingChanges: StreamChange[] = [];
+    if (missingMembers.length > 0) {
+      for (const member of missingMembers) {
+        try {
+          const dataStreamResult =
+            await this.dependencies.scopedClusterClient.asCurrentUser.indices.getDataStream({
```

Comment:
Can we parallelize this via `Promise.all` ? That would also allow us to report all missing streams, not just the first one

Reply:
I'll change this check + promote flow to use `Promise.all`, and I'll also update the validation below to report all missing Streams (meaning, those that were actually missing)